src/black/comments.py

```diff
@@ -88,7 +89,8 @@ def list_comments(prefix: str, *, is_endmarker: bool) -> List[ProtoComment]:
     form_feed = False
     for index, full_line in enumerate(re.split("\r?\n", prefix)):
         consumed += len(full_line) + 1  # adding the length of the split '\n'
-        line = full_line.lstrip()
+        match = re.match(r"^(\s*)(\S.*|)$", full_line)
+        whitespace, line = ("  ", full_line) if match is None else match.groups()
```

Comment:
The regex should always match, right? If so, would be clearer to add an assert.

Reply:
@JelleZijlstra Done